packages/web-components/src/components/locale-modal/locale-modal-composite.ts

```diff
@@ -181,32 +192,50 @@ class DDSLocaleModalComposite extends HybridRenderMixin(LitElement) {
                 ?invalid="${countryList.length === 0 ||
                 massagedCountryList?.find(({ region }) => region === name) ===
                   undefined}"
-                name="${name}"></dds-region-item>
+                name="${name}"
+                @click="${this._handleRegionClick}"></dds-region-item>
             `;
           })}
         </dds-regions>
-        <dds-locale-search
-          close-button-assistive-text="${ifNonNull(searchClearText)}"
-          label-text="${ifNonNull(searchLabel)}"
-          placeholder="${ifNonNull(searchPlaceholder)}"
-          availability-label-text="${ifNonNull(availabilityText)}"
-          unavailability-label-text="${ifNonNull(unavailabilityText)}">
-          ${massagedCountryList?.map(
-            ({ country, href, language, locale, region }) => html`
-              <dds-locale-item
-                country="${country}"
-                href="${href}"
-                language="${language}"
-                locale="${locale}"
-                region="${region}">
-              </dds-locale-item>
+
+        ${this.chosenRegionRender
+          ? html`
+              <dds-locale-search
+                close-button-assistive-text="${ifNonNull(searchClearText)}"
+                label-text="${ifNonNull(searchLabel)}"
+                placeholder="${ifNonNull(searchPlaceholder)}"
+                availability-label-text="${ifNonNull(availabilityText)}"
+                unavailability-label-text="${ifNonNull(unavailabilityText)}">
+                ${massagedCountryList
+                  ?.filter(({ region }) => {
+                    return region === this.chosenRegion;
+                  })
+                  .map(
+                    ({ country, href, language, locale, region }) => html`
+                      <dds-locale-item
+                        country="${country}"
+                        href="${href}"
+                        language="${language}"
+                        locale="${locale}"
+                        region="${region}">
+                      </dds-locale-item>
+                    `
+                  )}
+              </dds-locale-search>
             `
-          )}
-        </dds-locale-search>
+          : ``}
       </dds-locale-modal>
     `;
   }
 
+  /**
+   * Handles click on a region.
+   */
+  private _handleRegionClick(target) {
```

Comment:
Doesn't make any functional difference, but by convention the argument name for an event handler is typically `event` or `e` or something along those lines. Especially since there is a `target` property on it, it reads better to have it be `event.target`, similarily `event.currentTarget`.